### PR TITLE
New fuzzing documentation

### PR DESCRIPTION
**Pull request by deymo**
Add documentation about fuzzing and how to reproduce the fuzzing bugs.

### CODE REVIEW

File: doc/fuzzing.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+values past the end of the requested region. These values are part of the image
+padding but are not initialized. This behavior would not create an MSan error
+unless the processing includes operations like `NearestInt()`. For such cases
+the preferred solution is to use `msan::UnpoisonMemory` over the portion of
+memory of the last SIMD vector before processing, and then running
+`msan::PoisonMemory` over the corresponding value in the output side. A note
+including why this is safe to do must be added, for example if the processing
+doesn't involve any cross-lane computation.
+
+Initializing padding memory in MSan builds is discourage because it may hide
```
Comment (jonsneyers) [inline doc/fuzzing.md:154]: discouraged

File: doc/fuzzing.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+bugs in functions that weren't supposed to read from the padding. Initializing
+padding memory in all builds, including Release builds, would mitigate the
+MSan potential security issue but it would hide the logic bug for longer time
```
Comment (jonsneyers) [inline doc/fuzzing.md:157]: for **a** longer time

File: doc/fuzzing.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+   write.
+
+ * `JXL_DASSERT()` is only enabled in Debug builds, which includes all the ASan,
+   MSan and UBSan builds. Performance of these checks is not an issue if kept
+   within reasonable limits (automated msan/asan test should finish withing 1
+   hour for example). Fuzzing is more effective when the given input runs
+   faster, so keep that in mind when adding a complex DASSERT that runs multiple
+   times per output pixel.
+
+ * For MSan builds it is also possible to specify that certain value must be
```
Comment (jonsneyers) [inline doc/fuzzing.md:126]: certain value**s**

**Review by jonsneyers — DISMISSED** [review]
LGTM modulo typos.

File: doc/fuzzing.md (hunk 1 above)
Comment (deymo) [reply doc/fuzzing.md]: Done

File: doc/fuzzing.md (hunk 3 above)
Comment (deymo) [reply doc/fuzzing.md]: Done

**Comment by deymo** [comment]
I added a section about the env variables that make stack traces useful

**Review by jonsneyers — APPROVED** [review]

**Pull request merged**